.github/workflows/smoketest.yml

```diff
@@ -449,3 +450,84 @@ jobs:
+      run: |
+        sudo apt-get update
+        sudo apt-get install -y git wget net-tools
+        sudo apt-get install -y tzdata unzip xvfb libxi6
+        sudo apt-get install -y default-jdk
+        sudo apt-get -yq install bison flex build-essential binutils-dev libldap2-dev libcppunit-dev libicu-dev libxslt1-dev \
+          zlib1g-dev libboost-regex-dev libarchive-dev libv8-dev default-jdk libapr1-dev libaprutil1-dev libiberty-dev \
+          libhiredis-dev libtbb-dev libxalan-c-dev libnuma-dev nodejs libevent-dev libatlas-base-dev libblas-dev python3-dev \
+          default-libmysqlclient-dev libsqlite3-dev libmemcached-dev libcurl4-openssl-dev pkg-config libtool autotools-dev automake \
+          libssl-dev gdb
+        wget https://dl.google.com/linux/direct/google-chrome-stable_current_amd64.deb
+        sudo apt-get install -y ./google-chrome-stable_current_amd64.deb
+        wget https://chromedriver.storage.googleapis.com/2.41/chromedriver_linux64.zip
+        unzip chromedriver_linux64.zip
+        sudo mv chromedriver /usr/bin/chromedriver
+        sudo chown root:root /usr/bin/chromedriver
+        sudo chmod +x /usr/bin/chromedriver
+        wget https://selenium-release.storage.googleapis.com/3.141/selenium-server-standalone-3.141.59.jar
+        wget http://www.java2s.com/Code/JarDownload/testng/testng-6.8.7.jar.zip
+        unzip testng-6.8.7.jar.zip
+        export CLASSPATH=".:../selenium-server-standalone-3.141.59.jar:../testng-6.8.7.jar"
+
+    - name: run
+      run: |
+        export LANG="en_US.UTF-8"
+        sudo update-locale
+        tar --zstd -xf install.tgz
+        source install/opt/HPCCSystems/sbin/hpcc_setenv
+        install/opt/HPCCSystems/etc/init.d/hpcc-init start
+        mkdir tests
+        export CLASSPATH=".:../selenium-server-standalone-3.141.59.jar:../testng-6.8.7.jar"
+        if [[ -d install/test-ui/tests ]]
+        then
+                javaFilesCount=$(find install/test-ui/tests/ -iname '*.java' -type f -print | wc -l )
+                echo "Number of test java files is $javaFilesCount"
+                if [[ ${javaFilesCount} -ne 0 ]]
+                then
+                        # Copy testfiles into tests
+                        cp -v install/test-ui/tests/*java tests/
+                        cd tests
+                        javac *.java
+                        cd ..
+                        # get local IP
+                        LOCAL_IP4=$(ip -o -4 addr list eth0 | awk '{print $4}' | cut -d/ -f1)
+                        export LOCAL_IP4
+                        echo "LOCAL_IP4: $LOCAL_IP4"
+                        if [[ -f install/test-ui/run.sh ]]
+                        then
+                                cp -v install/test-ui/run.sh .
+                                ./run.sh http://$LOCAL_IP4:8010 > eclWatchUiTest.log 2>&1
+                                echo "UI test done"
+                                exit $( egrep -c '^Fail$' ./eclWatchUiTest.log )
```

Comment:
Is there any guarantee that "^Fail$" is part of the result?
It looks like run.sh is designed to return 0 on success and anything else should be used as the indicator of failure.
So think this should be changed to return error code from run.sh

Reply:
Changed